Devconf us 2024 report (#1)
FloMedja/medium_articles: devconf_us_aug_2024/devconf_us_aug_2024.md

a potential threat to our jobs. The current AI trend is similar to past groundbreaking technologies. We need to be mindful of the **social changes** 
brought about by technology and adopt a **systems thinking** approach to its implementation.

Some technology professions cannot fully automate their tasks. For instance, **product management** tools have not evolved as quickly as development tools. 
This can create bottlenecks in delivering enterprise software, potentially leading to unsatisfied customers and lost revenue.
AI tools can assist **product managers** by analyzing large datasets, automating workflows, and enhancing decision-making capabilities.

Jen highlights the importance of engineering and business teams collaborating to understand each other's processes and agreeing on shared workflows to streamline software delivery.
Some companies have achieved a change lead time of less than one day. High-performing teams are often the result of investments in development processes. 
Agile practices, DevOps transformations, and Cloud-Native development are a few examples of processes that help streamline the software development lifecycle.

> Finally Jen emphasizes that companies need to empower teams and invest in upskilling. 

Reviewer: You use past tense and present tense within the summary. 
Eg: "Jen spoke about AI.." vs "Jen highlights the importance of engineering.."

The talk with Kelsey Hightower is written in past tense but the other talks are written in present tense. I would suggest sticking to one tense throughout the article.

Reviewer 2: Fixed :)

## Talks

### [Building a Better Software Supply Chain](https://youtu.be/Ye6nyDcXE_8)

#### Takeaways: 
- **Never** commit your AWS Admin Key or Any Access Key to your git repository.
- **Automation** of cluster operations is valuable, even if it takes more time initially. 
  It is essential to implement automation and documentation for predictable and reproducible cluster creation.
- Backups are easy, but recovery can be challenging. It is crucial to perform practical scenarios to ensure that the backup/recovery process works as intended.
- SRE and platform engineering efforts can be more complex than anticipated. 
  Live clusters may require 24/7 infrastructure support, monitoring, and observability. Training SREs and platform engineers on the product can take several months.
- Penetration testing can help identify obvious vulnerabilities before going live.
- Onboarding teams to new technology takes time. Providing good documentation and code examples can significantly improve the onboarding process.

> This talk show the security risk that can happen during the process of delivery a solution to client.
> 
> The talk tells the story of the building of [Konflux](https://konflux-ci.dev/) by Red Hat. Konflux help developer builds a secure software supply chain process. It is a spin off of a Entreprise solution devolped at Red Hat.

Reviewer: ```suggestion
This talk shows the security risk that can happen during the process of delivery a solution to client. It tells the story of the building of [Konflux](https://konflux-ci.dev/) by Red Hat. Konflux help developer builds a secure software supply chain process. It is a spin off of a Entreprise solution devolped at Red Hat.
```

Reviewer 2: Fixed :)

> Ann Marie Fred talk about the challenges encouters during the devolpment of Konflux Enterprise version. 

Reviewer: ```suggestion
Ann Marie Fred talks about the challenges encoutered during the devolpment of Konflux Enterprise version. 
```

Reviewer 2: Fixed :)

### [Zero-instrumentation observability based on eBPF](https://youtu.be/xDrygKuR1ZM)

#### Takeaways
- There are four observability signals: logs, metrics, traces, and profiling.
- Each observability signal contributes to a different aspect of the observability system, and they complement one another.
- Instrumentation can be either static or dynamic and may vary in scope depending on the system.
- eBPF can be used to achieve observability without modifying a program's codebase.

Peter Zaitsev presents the use of various observability signals, including logs, metrics, traces, and profiling. 
He explains the purpose of each signal and how they contribute to system observability.

> [eBPF](https://ebpf.io/) allow to get visibility into many Linux kernels events. 
> The talk explains how [eBPF](https://ebpf.io/) programs can be used to have observability without modifying a program's codebase. 

Reviewer: ```suggestion
[eBPF](https://ebpf.io/) allows visibility into many Linux kernels events. The talk explains how [eBPF](https://ebpf.io/) programs can be used to achieve observability without modifying a program's codebase. 
```

Reviewer 2: Fixed :)

[Caroot](https://github.com/coroot/coroot) is a open source tool that can help add observability to a program.

### [ORAS: Powering the next generation of Cloud Native](https://youtu.be/RfMbKGWPwBw)

#### Takeaways
- OCI artifacts are an emerging standard for storing various types of artifacts for container workloads.
- OCI artifacts can benefit from the same processes and software lifecycle as OCI images.

Andrew Block presents the container runtime specification and explains the different components of a container image.

> An OCI image consists of an image manifest, an optional image index, a set of filesystem layers, and a configuration. A container is a running instance of this image. Deep down, containers are files and processes.
> 
> The OCI image manifest defines the media type for the image and its layers.

Reviewer: ```suggestion
An OCI image consists of an image manifest, an optional image index, a set of filesystem layers, and a configuration. A container is a running instance of this image. Deep down, containers are files and processes. The OCI image manifest defines the media type for the image and its layers.
```

Reviewer 2: Fixed :)

> OCI supports storing arbitrary content types called **artifacts**, enabling the storage and serving of content types other than container images.
> 
> OCI Artifacts can store:

Reviewer: ```suggestion
OCI supports storing arbitrary content types called **artifacts**, enabling the storage and serving of content types other than container images.
OCI Artifacts can store:
```

Reviewer 2: Fixed :)

- Software Bill Of Materials (SBOM)
- Signatures
- Software packages (.jar,rpm)
- AI/ML models

> The benefict of OCI Artifacts :

Reviewer: ```suggestion
Benefits of OCI Artifacts:
```

Reviewer 2: Fixed :)
